staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go

```diff
@@ -576,19 +573,22 @@ func TestGetHistogramVecFromGatherer(t *testing.T) {
 				Buckets:   buckets,
 			}
 			vec := metrics.NewHistogramVec(HistogramOpts, labels)
-			registerMetrics.Do(func() {
-				legacyregistry.MustRegister(vec)
-			})
+			// Use local registry
+			var registry = metrics.NewKubeRegistry()
+			var gather metrics.Gatherer = registry
+			registry.MustRegister(vec)
```

Comment:
nit: `defer registry.Reset()` here is a bit cleaner

Reply:
`Defer` inside for loop is a little tricky. 

It will begin to`Reset` after all loops finish instead of one loop finishes.
https://play.golang.org/p/9e9qyuE64nI

Reply:
It's wrapped in a function `func(t *testing.T)`, so your playground isn't an accurate representation of the flow. Defers are local to the function it is invoked in, so using a defer here should be fine.

Reply:
Oh, you're right.
I tested it.

https://gist.github.com/CatherineF-dev/3ca9d99aefda22775570cec7edb493af
```
=== RUN   TestGetHistogramVecFromGatherer
    metrics_test.go:564: ## begin main
=== RUN   TestGetHistogramVecFromGatherer/filter_with_one_label
    metrics_test.go:568: ## begin for
    metrics_test.go:571: ## end for
## defer 0
=== RUN   TestGetHistogramVecFromGatherer/filter_with_two_labels
    metrics_test.go:568: ## begin for
    metrics_test.go:571: ## end for
## defer 1
## end main
=== CONT  TestGetHistogramVecFromGatherer
    metrics_test.go:602: ## end main
```